src/shipit_static_analysis/shipit_static_analysis/clang.py

```diff
@@ -43,6 +44,27 @@
     '--from-build', 'linux64-clang-tidy'
 ]
 
+__third_party = None
+
+
+def list_third_party(work_dir, path='tools/rewriting/ThirdPartyPaths.txt'):
+    '''
+    List third party directories using mozilla-central file
+    Uses a local cache to avoid reading every time the file
```

Comment:
Fair enough, but I'm still not convinced that this local cache really is an optimization (as opposed to what the file system already does by default), and that it won't cause stale cache issues (I'm not sure if this cache is refreshed once per push or once per commit, but if it's once per push, we'll miss any `ThirdPartyPaths.txt` changes that are not in the first commit; and if it's once per commit, why cache the value if it is only ever read once?)